client/maps/action-creators.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  GetMapDetailsPayload,
+  GetMapsPayload,
+  MapInfoJson,
+  MapPreferences,
+  MapSortType,
+  MapVisibility,
+  Tileset,
+  UpdateMapPayload,
+  UpdateMapServerBody,
+} from '../../common/maps'
+import { ThunkAction } from '../dispatch-registry'
+import fetch from '../network/fetch'
+import { apiUrl } from '../network/urls'
+import { openSnackbar } from '../snackbars/action-creators'
+import { ClearMaps } from './actions'
+
+const upload = IS_ELECTRON ? require('./upload').default : null
+
+export function uploadLocalMap(path: string, onMapSelect: (map: MapInfoJson) => void): ThunkAction {
+  return async dispatch => {
+    if (!upload) {
+      return
+    }
+
+    dispatch({
+      type: '@maps/uploadLocalMapBegin',
+      payload: { path },
+    })
+
+    dispatch({
+      type: '@maps/uploadLocalMap',
+      payload: upload(path, apiUrl`maps`).then(({ map }: { map: MapInfoJson }) => {
```

Comment:
You added a generic parameter to this, so you should use that instead of typing things in the `then` (and ideally the type specified would be a Payload type in `common/` that the server uses as well)